src/compact_x25519.c

```diff
@@ -21,7 +21,12 @@ void compact_x25519_shared(
     const uint8_t my_private_key[X25519_KEY_SIZE], 
     const uint8_t their_public_key[X25519_KEY_SIZE]
 ) {
-    c25519_smult(shared_secret, their_public_key, my_private_key);
+    // Ensure that supplied private key is clamped (fix issue #1).
+    // Applying `c25519_prepare` multiple times to same private key is fine.
+    uint8_t clamped_private_key[X25519_KEY_SIZE];
```

Comment:
this array now contains the private key, we don't want to leave this lying around outside of the control of the caller.

It should be `compact_wipe` after usage.

Reply:
That's reasonable. Applied.